ccdproc/image_collection.py

```diff
@@ -691,14 +691,11 @@ def _generator(self, return_type,
 
             file_name = path.basename(full_path)
 
-            if 'unit' not in ccd_kwargs:
-                ccd_kwargs['unit'] = 'adu'
-
-            return_options = {'header': hdulist[0].header,
-                              'hdu': hdulist[0],
-                              'data': hdulist[0].data,
-                              'ccd': fits_ccddata_reader(file_name,
-                                                         **ccd_kwargs)}
+            return_options = {'header': lambda: hdulist[0].header,
+                              'hdu': lambda: hdulist[0],
+                              'data': lambda: hdulist[0].data,
+                              'ccd': lambda: fits_ccddata_reader(file_name,
```

Comment:
Since you change all these lines, you can also change the indentation here to avoid the last line break:

```
            return_options = {
                'header': lambda: hdulist[0].header,
                'hdu': lambda: hdulist[0],
                'data': lambda: hdulist[0].data,
                'ccd': lambda: fits_ccddata_reader(full_path, **ccd_kwargs)
                }
```